src/visitor/RewriteAllSubExprVisitor.h

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef VISITOR_REWRITEALLSUBEXPRVISITOR_H_
+#define VISITOR_REWRITEALLSUBEXPRVISITOR_H_
+
+#include "common/expression/Expression.h"
+#include "visitor/ExprVisitorImpl.h"
+
+namespace nebula {
+namespace graph {
+
+class RewriteAllSubExprVisitor final : public ExprVisitorImpl {
+public:
+    explicit RewriteAllSubExprVisitor(const Expression* rewrittenSubExpr,
+                                      const Expression* rewriteSubExpr,
+                                      Expression* originExpr);
+
```

Comment:
Prefer a `RewriteVarExprVisitor` to replace `$var` expression with _new expression_. Do you check wether these existing RewriteVisitors meet your requirements, such as `RewriteLabelAttrVisitor` or `RewriteSymExprVisitor` ?

Reply:
I have considered the existing visitor, none of them can meet the needs here. I hope to implement a more general visitor, which will rewrite all internal expressions equal to `rewrittenSubExpr` in a certain parrent expression(originExpr here) to `rewriteSubExpr`, and It's not just about judging the type of expression. Furthermore, VariablePropertyExpr here is just a special case, and this is a type-independent visitor.

Reply:
I plan to abstract and unify all rewrite visitors, and the visitor here can also be deleted after completion.
Refer to [PR#870](https://github.com/vesoft-inc/nebula-graph/pull/870) for details about visitor refactoring.

Reply:
Done.